### PR TITLE
Adds Slappers

### PR DESCRIPTION
**Pull request by deathride58**
It just needs sprites.

:cl: deathride58
Add: *slap
/:cl:

*slap to ready your slappers.
Help/Disarm/Grab intent to harmlessly slap. Harm to slap for 1 brute.

### CODE REVIEW

**Comment by calzilla1** [comment]
God damn it now we need a purple leopard skin suit and hat that lets you back hand honkies for 10 brute and a short stun

File: code/game/objects/items/weaponry.dm [hunk 1]
```diff
@@ -586,6 +586,34 @@
...
+	if(user.a_intent != INTENT_HARM)
+		var/aim_for_face = ((user.zone_selected == "mouth") || (user.zone_selected == "eyes"))
+		user.do_attack_animation(M)
+		playsound(M.loc, 'sound/weapons/slap.ogg', 50, 1, -1)
+		user.visible_message("<span class='danger'>[user] slaps [M] in the [(aim_for_face)?"face":user.zone_selected]!</span>",
+ 		"<span class='notice'>You slap [M] in the [(aim_for_face)?"face":user.zone_selected]! </span>",\
+ 		"You hear a slap.")
+		if(ishuman(M))
+			var/mob/living/carbon/human/L = M
+			L.endTailWag()
```
Comment (Jordie0608) [inline code/game/objects/items/weaponry.dm]: Is this supposed to be here?

Comment (deathride58) [reply code/game/objects/items/weaponry.dm]: Yeah

Comment (The-clown-known-as-Pepe) [reply code/game/objects/items/weaponry.dm]: its copypasted code from slapping feature

It was originally made to stop lizards wagging their tails.

File: code/game/objects/items/weaponry.dm [hunk 2]
```diff
@@ -586,6 +586,34 @@
...
 
+/obj/item/slapper
+	name = "slapper"
+	desc = "This is how real men fight."
+	icon_state = "latexballon"
+	force = 1
+	throwforce = 0
+	flags_1 = DROPDEL_1 | ABSTRACT_1
+	attack_verb = list("slapped")
+	hitsound = 'sound/effects/snap.ogg'
```
Comment (kevinz000) [inline code/game/objects/items/weaponry.dm]: we used to have slapping is this the right sound file?

File: code/modules/mob/living/emote.dm [hunk 3]
```diff
@@ -513,3 +513,17 @@
...
+	key_third_person = "slaps"
+	restraint_check = TRUE
+
+/datum/emote/living/slap/run_emote(mob/user, params)
+	. = ..()
+	var/obj/item/slapper/N = new(user)
+	if(user.put_in_hands(N))
+		to_chat(user, "<span class='notice'>You ready your slapping hand.</span>")
+	else
+		qdel(N)
```
Comment (Cyberboss) [inline code/modules/mob/living/emote.dm]: DROPDEL will already qdel it

File: code/game/objects/items/weaponry.dm [hunk 4]
```diff
@@ -586,6 +586,34 @@
...
+ 		"<span class='notice'>You slap [M] in the [(aim_for_face)?"face":user.zone_selected]! </span>",\
+ 		"You hear a slap.")
+		if(ishuman(M))
+			var/mob/living/carbon/human/L = M
+			L.endTailWag()
+		return
+	else
+		if(ishuman(M))
+			var/mob/living/carbon/human/L = M
+			L.endTailWag()
```
Comment (Cyberboss) [inline code/game/objects/items/weaponry.dm]: Duplicated code

File: code/game/objects/items/weaponry.dm [hunk 5]
```diff
@@ -586,6 +586,34 @@
...
+	throwforce = 0
+	flags_1 = DROPDEL_1 | ABSTRACT_1
+	attack_verb = list("slapped")
+	hitsound = 'sound/effects/snap.ogg'
+
+/obj/item/slapper/attack(mob/M, mob/living/carbon/human/user)
+	if(user.a_intent != INTENT_HARM)
+		var/aim_for_face = ((user.zone_selected == "mouth") || (user.zone_selected == "eyes"))
+		user.do_attack_animation(M)
+		playsound(M.loc, 'sound/weapons/slap.ogg', 50, 1, -1)
```
Comment (Cyberboss) [inline code/game/objects/items/weaponry.dm]: playsound(M, ...

File: code/game/objects/items/weaponry.dm [hunk 6]
```diff
@@ -586,6 +586,34 @@
...
+	icon_state = "latexballon"
+	force = 1
+	throwforce = 0
+	flags_1 = DROPDEL_1 | ABSTRACT_1
+	attack_verb = list("slapped")
+	hitsound = 'sound/effects/snap.ogg'
+
+/obj/item/slapper/attack(mob/M, mob/living/carbon/human/user)
+	if(user.a_intent != INTENT_HARM)
+		var/aim_for_face = ((user.zone_selected == "mouth") || (user.zone_selected == "eyes"))
```
Comment (Cyberboss) [inline code/game/objects/items/weaponry.dm]: head too

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weaponry.dm (hunk 2 above)
Comment (Cyberboss) [reply code/game/objects/items/weaponry.dm]: yes

**Comment by YakumoChen** [comment]
Head pats too?????

**Comment by deathride58** [comment]
Out of this PR's scope.

**Comment by Bawhoppen** [comment]
@YakumoChen This PR isn't adding actual slaps back, it's adding some item to slap with

**Comment by ExcessiveUseOfCobblestone** [comment]
@Bawhoppen did you care to read the code?

It's the equivalent to instead of having help intent on head for pats you say *pat and you get a little pat hand that now pats heads and go "_Actually_ it's an item so it's not actually adding pats back"

**Comment by Bawhoppen** [comment]
I skimmed it but that seems like a weird way to do it

**Comment by PKPenguin321** [comment]
This is what I did for the circle game, it should be fine

**Review by Cyberboss — APPROVED** [review]

**Comment by optimumtact** [comment]
This should not have been merged as is

**Comment by Bawhoppen** [comment]
Really shouldn't have been

**Comment by optimumtact** [comment]
like this is literally why we added ranged targeting to spell effects

**Comment by deathride58** [comment]
aside from sprites how is this broken

**Comment by optimumtact** [comment]
it's not broken, it's using the wrong systems to achieve it's goal

**Comment by Bawhoppen** [comment]
sprite is broken too

**Pull request merged**